### ISSUE TITLE
`map_elements` fails to throw an exception when returning a struct or list

### ISSUE THREAD

**Issue by edwinvehmaanpera** [pola-rs/polars#14821]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example
_Updated minimal:_
```python
# skip_nulls=True: Swallows the exception and returns Null
pl.Series(["2", "e"]).map_elements(lambda x: [int(x)], skip_nulls=True)
#shape: (2,)
#Series: '' [list[i64]]
#[
#        [2]
#        null
#]
```
Original:
```python
data = {"a": ["2", "e"], "ord": [2,1]}

def convert_to_int(x):
    return {"field": int(x)}

df = pl.DataFrame(data)

# Having failure case first, fails as expected with "ValueError: invalid literal for int() with base 10: 'e'"
df = df.sort("ord").with_columns(b=pl.col("a").map_elements(convert_to_int, return_dtype=pl.Struct({'field': pl.Int64})))

# When failure case is not first, map_elements doesn't raise exception, instead returns null: 
#     shape: (2, 3)
#     ┌─────┬─────┬───────────┐
#     │ a   ┆ ord ┆ b         │
#     │ --- ┆ --- ┆ ---       │
#     │ str ┆ i64 ┆ struct[1] │
#     ╞═════╪═════╪═══════════╡
#     │ 2   ┆ 2   ┆ {2}       │
#     │ e   ┆ 1   ┆ {null}    │
#     └─────┴─────┴───────────┘
df = df.with_columns(b=pl.col("a").map_elements(convert_to_int, return_dtype=pl.Struct({'field': pl.Int64})))


# When returning an integer, order does not matter. Both fail: "ValueError: invalid literal for int() with base 10: 'e'"
df = df.with_columns(b=pl.col("a").map_elements(int, return_dtype=pl.Int64))
df = df.sort("ord").with_columns(b=pl.col("a").map_elements(int, return_dtype=pl.Int64))
```

### Issue description

Using map_elements and an UDF which returns a struct seems not to raise an exception when the exception case is not the first item. It instead returns "{null}". 

The example is obviously just an arbitary example of this behaviour.

### Expected behavior

Returning a struct should not cause inconsistent behaviour and the UDF in this case should raise an exception.

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.13
Index type:           UInt32
Platform:             Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
Python:               3.11.4 (main, Jul  5 2023, 13:45:01) [GCC 11.2.0]

----Optional dependencies----
adbc_driver_manager:  0.10.0
cloudpickle:          3.0.0
connectorx:           0.3.2
deltalake:            0.15.3
fsspec:               2023.12.2
gevent:               24.2.1
hvplot:               0.9.2
matplotlib:           3.8.3
numpy:                1.26.4
openpyxl:             3.1.2
pandas:               2.2.0
pyarrow:              15.0.0
pydantic:             2.6.1
pyiceberg:            0.6.0
pyxlsb:               1.0.10
sqlalchemy:           2.0.27
xlsx2csv:             0.8.1
xlsxwriter:           3.2.0
```

</details>

**Comment by edwinvehmaanpera** [comment]
Tested against 1.2.1 and this is still an issue. Would it be possible for someone to take a look?

The random nature and unexpected nulls of this issue means that using map_elements inconjunction with functions that return structs seems currently unusable.

**Comment by cmdlineluser** [comment]
It seems this also happens with lists.

```python
# NOT OK: Exception swallowed

pl.DataFrame({"a": ["2", "e"]}).select(pl.col("a").map_elements(lambda x: [int(x)]))
# shape: (2, 1)
# ┌───────────┐
# │ a         │
# │ ---       │
# │ list[i64] │
# ╞═══════════╡
# │ [2]       │
# │ null      │
# └───────────┘
```

```python
pl.DataFrame({"a": ["2", "e"]}).select(pl.col("a").map_elements(lambda x: int(x)))
# PanicException: python function failed ValueError: invalid literal for int() with base 10: 'e'
```

It may need @ritchie46 to take a look.

**Comment by edwinvehmaanpera** [comment]
I was hopeful that fixes related to https://github.com/pola-rs/polars/issues/18472 would fix this, but unfortunately this issue still exists in 1.7.1. However, it is somehow related to `skip_nulls` as:

```python
# skip_nulls=True: Swallows the exception and returns Null
pl.Series(["2", "e"]).map_elements(lambda x: [int(x)], skip_nulls=True)
#shape: (2,)
#Series: '' [list[i64]]
#[
#        [2]
#        null
#]
```
```python
# skip_nulls=False: Panics and raises an exception
pl.Series(["2", "e"]).map_elements(lambda x: [int(x)], skip_nulls=False)
# pyo3_runtime.PanicException: python function failed ValueError: invalid literal for int() with base 10: 'e'
```

**Comment by cmdlineluser** [comment]
It also raises if you set the `return_dtype=`

```python
pl.Series(["2", "e"]).map_elements(lambda x: [int(x)], return_dtype=pl.Int64)
# PanicException: python function failed ValueError: invalid literal for int() with base 10: 'e'
```

I have traced your example down to this point.

https://github.com/pola-rs/polars/blob/71a8b0543362aadb771c2a446ac93756ea36a124/crates/polars-python/src/map/series.rs#L236-L237

But If I add a `dbg!()` on line 238 it doesn't execute.

`call1` appears to be a `pyo3` method - so I'm not sure what is happening here or how to debug any further.

**Comment by cmdlineluser** [comment]
This is also fixed by https://github.com/pola-rs/polars/issues/19315 

```python
pl.Series(["2", "e"]).map_elements(lambda x: [int(x)], skip_nulls=True)
# ValueError: invalid literal for int() with base 10: 'e'
```

**Comment by edwinvehmaanpera** [comment]
Yes, thanks @cmdlineluser. This is now fixed in 1.18.0.
